libraries/chain_kv/include/b1/session/cache.hpp

```diff
@@ -0,0 +1,438 @@
+#pragma once
```

Comment:
Originally I implemented this type because of two reasons:
1.  Circular iterators.  `std::map` doesn't support circular iterators.  Which leads into the second reason...
2.  A common interface of all types in this design that store data.  I can cache, session, rocks_sessions interchangeably and in a generic fashion.  There was originally discussion about using `std::map`, 'std::unordered_map' or whatever container was discovered to be the best after performance tests. I wanted to insulate Session from the mess of changing between those types.  Ultimately when we discover the best backing container after performance testing, we can merge the container directly into session.

While I ultimately ended up with implementing the circular iterator logic in Session (and specializations of Session), the second reason still holds.